### PR TITLE
Allow for only a subset of the rule to be compiled #12

### PR DESCRIPTION
**Pull request by lucamolteni**
This solves https://github.com/kiegroup/ruleops/issues/12

Split modules:
* ruleops-basic (default sanity rules)
* ruleops-core (library)
* rhoc (RHOC specific rules)

### CODE REVIEW

File: rhoc/pom.xml [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        <type>pom</type>
+        <scope>import</scope>
+      </dependency>
+    </dependencies>
+  </dependencyManagement>
+  <dependencies>
+    <dependency>
+      <groupId>org.drools.ruleops</groupId>
+      <artifactId>ruleops-core</artifactId>
+      <version>1.0.0-SNAPSHOT</version>
```
Comment (lucamolteni) [inline rhoc/pom.xml:35]: Shall we add the ruleops version inside the parent pom?

File: ruleops-basic/pom.xml [hunk 2]
```diff
@@ -35,6 +21,12 @@
     </dependencies>
   </dependencyManagement>
   <dependencies>
+    <dependency>
+      <groupId>org.drools.ruleops</groupId>
+      <artifactId>ruleops-core</artifactId>
+      <version>1.0.0-SNAPSHOT</version>
```
Comment (lucamolteni) [inline ruleops-basic/pom.xml:27]: Same here, we should use a property for the version

File: rhoc/pom.xml [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
```
Comment (tarilabs) [inline rhoc/pom.xml:29]:
```suggestion
      <version>${project.version}</version>
```
?

File: rhoc/pom.xml [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.drools.ruleops</groupId>
+    <artifactId>ruleops-parent</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+  </parent>
+
+  <artifactId>rhoc</artifactId>
```
Comment (tarilabs) [inline rhoc/pom.xml:12]:
```suggestion
  <artifactId>ruleops-rhoc</artifactId>
```

?

File: ruleops-basic/pom.xml (hunk 2 above)
Comment (tarilabs) [reply ruleops-basic/pom.xml:27]: Same suggestion as above?

**Review by tarilabs — COMMENTED** [review]
There is a file `ruleops-core/src/main/resources/META-INF/beans.xml` which I'm not sure why is needed?

To me this is a good iteration on the original topic (#12) but further iteration could consider having the "core" as a quarkus extension.

Such as, we have "a rule base".

1. mvn quarkus-plugin including ruleops extension
2. copy "rule base" inside
3. done.

that would help codegenerate the main version with meaningful help/options (by inspecting the rulebase in the extension)

Same consideration for the topic of https://github.com/kiegroup/ruleops/issues/23

I'd merge this one and iterate according to the above ^ later

File: rhoc/pom.xml (hunk 4 above)
Comment (lucamolteni) [reply rhoc/pom.xml:12]: I'd rather keep "rules only" modules outside of RuleOps, eventually I think they should be moved to another project as well

**Comment by lucamolteni** [comment]
> There is a file `ruleops-core/src/main/resources/META-INF/beans.xml` which I'm not sure why is needed?

That's to define beans in an external module, otherwise you get a ```[ERROR] 	Suppressed: javax.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type org.drools.ruleops.DroolsSingleton and qualifiers [@Default]```

**Comment by lucamolteni** [comment]
> To me this is a good iteration on the original topic (#12) but further iteration could consider having the "core" as a quarkus extension.
> 
> Such as, we have "a rule base".
> 
> 1. mvn quarkus-plugin including ruleops extension
> 2. copy "rule base" inside
> 3. done.
> 
> that would help codegenerate the main version with meaningful help/options (by inspecting the rulebase in the extension)
> 
> Same consideration for the topic of #23
> 
> I'd merge this one and iterate according to the above ^ later

Created issue https://github.com/kiegroup/ruleops/issues/28

**Review by tarilabs — APPROVED** [review]

File: rhoc/pom.xml [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>${quarkus.platform.group-id}</groupId>
+        <artifactId>${quarkus.platform.artifact-id}</artifactId>
+        <version>${quarkus.platform.version}</version>
+        <type>pom</type>
+        <scope>import</scope>
+      </dependency>
+    </dependencies>
```
Comment (danielezonca) [inline rhoc/pom.xml:23]: I would move this directly to the root `pom.xml` instead of here (and in the `ruleops-core` module)

Comment (lucamolteni) [reply rhoc/pom.xml]: Done

File: rhoc/README.md [hunk 6]
```diff
@@ -0,0 +1,11 @@
+Compile the RHOC SOP Jar with 
+
+```shell
+mvn install
+```
+
+Then you can run the only RHOC SOP defined
+
+```shell
+java -jar target/quarkus-app/quarkus-run.jar cckp06od86599sf6h3fg
```
Comment (danielezonca) [inline rhoc/README.md:10]:
```suggestion
java -jar target/quarkus-app/quarkus-run.jar $RHOC_CLUSTER_ID
```

**Review by danielezonca — APPROVED** [review]

**Pull request merged**